### PR TITLE
Alerting docs: `Provision alerting resources` updates

### LINKED ISSUE #82005: DevRel/Docs: update provisioning docs
### What happened?

Updates provisioning docs as per https://docs.google.com/document/d/135XT70bZL3PUo20dLnvVY34S8-An7LAfVTEb6ylY_nw/edit#heading=h.smgyye4qizii

### What did you expect to happen?

x

### Did this work before?

x

### How do we reproduce it?

1.
2.
3.
x

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana:
OS:
Browser:

### Grafana platform?

None

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by ppcano**
Closes: #82005 

- Revise the structure and content across all `Provision alerting` documentation for clarity and consistency.
- Use `Provision` in titles and menu titles to be consistent with [Provision Grafana](https://grafana.com/docs/grafana/latest/administration/provisioning/). Use `Import` for headings.
- Include `Alerting Provisioning API` to the left-had sidebar.
- Minor cosmetic changes and fixes as per this [internal doc#Export/Import/Provisioning](https://docs.google.com/document/d/135XT70bZL3PUo20dLnvVY34S8-An7LAfVTEb6ylY_nw/edit#heading=h.c8kcmsqr1qic).

### CODE REVIEW

File: docs/sources/alerting/set-up/provision-alerting-resources/_index.md [hunk 1]
```diff
@@ -14,128 +14,68 @@ labels:
...
-To modify imported alert rules, you can use the **Modify export** feature to edit and then export.
+## Import alerting resources
+
+Choose from the options below to import (or provision) your Grafana Alerting resources.
 
-Choose from the options below to import your Grafana Alerting resources.
+1. [Use configuration files to provision your alerting resources][alerting_file_provisioning], such as alert rules and contact points, through files on disk.
 
-1. Use file provisioning to manage your Grafana Alerting resources, such as alert rules and contact points, through files on disk.
+   **Note:**
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/provision-alerting-resources/_index.md:35]: you can remove this Note - the admonition short code type note takes care of all that for you:)

File: docs/sources/alerting/set-up/provision-alerting-resources/_index.md [hunk 2]
```diff
@@ -14,128 +14,68 @@ labels:
...
-}
-```
-
-3. Use [Terraform](https://www.terraform.io/).
+
+2. Use [Terraform to provision alerting resources][alerting_tf_provisioning].
+
+3. Use the [Alerting provisioning HTTP API][alerting_http_provisioning] to manage alerting resources.
+
+   **Note:**
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/provision-alerting-resources/_index.md:43]: use the admonition shortcode here for note

File: docs/sources/alerting/set-up/provision-alerting-resources/http-api-provisioning/_index.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  - grafana
+  - alerting
+  - alerting resources
+  - provisioning
+labels:
+  products:
+    - cloud
+    - enterprise
+    - oss
+title: Use the HTTP API to manage alerting resources
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/provision-alerting-resources/http-api-provisioning/_index.md:14]: just a question on choice of using manage here - why not provision like the other two?

File: docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md [hunk 4]
```diff
@@ -11,41 +11,44 @@ keywords:
...
 weight: 200
 ---
 
-# Use Terraform to manage alerting resources
+# Use Terraform to provision alerting resources
 
 Use Terraform’s Grafana Provider to manage your alerting resources and provision them into your Grafana system. Terraform provider support for Grafana Alerting makes it easy to create, manage, and maintain your entire Grafana Alerting stack as code.
 
-For more information on managing your alerting resources using Terraform, refer to the [Grafana Provider](https://registry.terraform.io/providers/grafana/grafana/latest/docs) documentation.
+For more examples and information about the Terraform Alerting schemas, refer to the [Grafana Provider](https://registry.terraform.io/providers/grafana/grafana/latest/docs) documentation.
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md:26]:
```suggestion
Refer to [Grafana Provider](https://registry.terraform.io/providers/grafana/grafana/latest/docs) documentation for more examples and information on Terraform Alerting schemas.
```

File: docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md [hunk 5]
```diff
@@ -11,41 +11,44 @@ keywords:
...
 
 Complete the following tasks to create and manage your alerting resources using Terraform.
 
 1. Create an API key for provisioning.
-1. Configure the Terraform provider.
-1. Define your alerting resources in Terraform.
-1. Run `terraform apply` to provision your alerting resources.
+2. Configure the Terraform provider.
+3. Define your alerting resources in Terraform. [Export alerting resources][alerting_export] in Terraform format, or implement the [Terraform Alerting schemas](https://registry.terraform.io/providers/grafana/grafana/latest/docs).
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md:32]:
```suggestion
1. Configure the Terraform provider.
1. Define your alerting resources in Terraform. [Export alerting resources][alerting_export] in Terraform format, or implement the [Terraform Alerting schemas](https://registry.terraform.io/providers/grafana/grafana/latest/docs).
```

Comment (brendamuir) [reply docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md:32]: we use lazy numbering in markdown is always 1. 1. 1. and it renders correctly https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#ordered-lists

File: docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md [hunk 6]
```diff
@@ -155,209 +156,228 @@ EOT
...
 
 2. In the mute_timings field, link a mute timing to your notification policy.
 
-3. Run the command ‘terraform apply’.
+3. Run the command `terraform apply`.
 
 4. Go to the Grafana UI and check the details of your notification policy.
 
-**Note:**
+   **Note:**
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md:209]: Use admonition for note {{< admonition type="note" >}}
{{< /admonition >}}

Comment (brendamuir) [reply docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md:209]: and lazy numbering for all please:)

**Comment by brendamuir** [comment]
Also -- we probably want to consider backporting these changes - back to the last major version - so 10.

File: docs/sources/alerting/set-up/provision-alerting-resources/http-api-provisioning/_index.md (hunk 3 above)
Comment (ppcano) [reply docs/sources/alerting/set-up/provision-alerting-resources/http-api-provisioning/_index.md:14]: First, I found the **The HTTP API and File Provisioning use distinct formats and aren’t interchangeable** --> https://github.com/grafana/grafana/issues/81387

Then, I clarified this on this PR: 

>    The JSON output from the majority of Alerting HTTP endpoints is not compatible for provisioning via configuration files. Instead, use the [Export Alerting endpoints][alerting_export_http] to return or download the alerting resources in provisioning format.

and

> Note that most Alerting endpoints return a JSON format that is not compatible for provisioning via configuration files, except the ones listed below.

Why did I use `manage` for the HTTP API?

I interpret Provisioning as setting up and managing the infrastructure components in an automated way via Code during the initialization process. With the API, users can manage individual components directly and be used for other cases (not only Provisioning)

I am unsure about this. And perhaps, this distinction makes it more complicate. Would be good to more feedback.

Comment (brendamuir) [reply docs/sources/alerting/set-up/provision-alerting-resources/http-api-provisioning/_index.md:14]: ok definitely something for the wider team then

File: docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md (hunk 6 above)
Comment (ppcano) [reply docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md:211]: fixed by https://github.com/grafana/grafana/pull/82221/commits/abbd39ca219b9e8d137ae46320a4fc0ee7c15456

File: docs/sources/alerting/set-up/provision-alerting-resources/_index.md (hunk 2 above)
Comment (ppcano) [reply docs/sources/alerting/set-up/provision-alerting-resources/_index.md:43]: fixed by https://github.com/grafana/grafana/pull/82221/commits/abbd39ca219b9e8d137ae46320a4fc0ee7c15456

Thanks to @jdbaldry

File: docs/sources/alerting/set-up/provision-alerting-resources/_index.md (hunk 1 above)
Comment (ppcano) [reply docs/sources/alerting/set-up/provision-alerting-resources/_index.md:35]: fixed by https://github.com/grafana/grafana/pull/82221/commits/abbd39ca219b9e8d137ae46320a4fc0ee7c15456

**Review by brendamuir — APPROVED** [review]
lgtm!

**Pull request merged**